### PR TITLE
code update to store models locally

### PR DESCRIPTION
**Pull request by AshishMahendra**
the list of modules updated
text_seg,
use_qa,
use_enc,
bi_enc,
cl_summer,
ent_ext,
tfm_ner

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,6 +32,7 @@ mydatabase
 celerybeat-schedule*
 node_modules
 *.jaseci_logs/
+_jaseci_ai_kit
```
Comment (ypkang) [inline .gitignore]: Is this needed?

File: jaseci_ai_kit/jac_nlp/jac_nlp/bi_enc/bi_enc.py [hunk 2]
```diff
@@ -307,8 +317,9 @@ def save_model(model_path: str):
                 Invalid model name. Model Name can only have Alphanumeric
                  and '_' characters.""",
             )
-        if not os.path.exists(model_path):
-            os.makedirs(model_path)
+        if not os.path.isabs(model_path):
+            model_path = str(BI_ENC_ROOT / Path(model_path))
```
Comment (ypkang) [inline jaseci_ai_kit/jac_nlp/jac_nlp/bi_enc/bi_enc.py]: I think `os.path.join` is a more robust way to do this than `str`

File: jaseci_ai_kit/jac_nlp/jac_nlp/bi_enc/bi_enc.py [hunk 3]
```diff
@@ -35,17 +35,27 @@ def setup():
     Loading configurations from utils/config.cfg and
     initialize tokenizer and model
     """
-    global model, tokenizer, model_config, train_config, t_config_fname, m_config_fname
+    global model, tokenizer, model_config, train_config, t_config_fname, m_config_fname, BI_ENC_ROOT  # noqa
```
Comment (ypkang) [inline jaseci_ai_kit/jac_nlp/jac_nlp/bi_enc/bi_enc.py]: Let's have `BI_ENC_ROOT` at the top level of this file/module. Its not really a global variable. It is a constant.

Comment (ypkang) [reply jaseci_ai_kit/jac_nlp/jac_nlp/bi_enc/bi_enc.py]: Same with the other modules too.

File: jaseci_ai_kit/jac_nlp/jac_nlp/ent_ext/ent_ext.py [hunk 4]
```diff
@@ -51,12 +57,16 @@ def setup(reload=False):
...
-        tagger = SequenceTagger.load(NER_MODEL_NAME)
+        tagger = SequenceTagger.load(cached_path(NER_MODEL_NAME, MODEL_PATH))
     elif MODEL_TYPE.lower() in "tars" and NER_MODEL_NAME.lower() != "none":
-        tagger = TARSTagger.load(NER_MODEL_NAME)
+        tagger = TARSTagger.load(cached_path(TARS_NER_PATH, MODEL_PATH))
     print(f"loaded mode : [{NER_MODEL_NAME}]")
 
 
+# initialize the tagger
+setup()
```
Comment (ypkang) [inline jaseci_ai_kit/jac_nlp/jac_nlp/ent_ext/ent_ext.py]: Remember to remove this, as we have discussed.

**Review by ypkang — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (AshishMahendra) [reply .gitignore]: actually, I have a backup folder that has some of the bi_encoder results.

**Pull request closed without merging**